google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionImpl.java

```diff
@@ -463,7 +446,7 @@ TransactionContextImpl newTransaction(Options options) {
     return TransactionContextImpl.newBuilder()
         .setSession(this)
         .setOptions(options)
-        .setTransactionId(readyTransactionId)
+        .setTransactionId(null)
```

Comment:
`readyTransactionId` was only being set in prepareReadWriteTransaction() method. In absence of it, this will always remain null. @olavloite could you give this a closer look if this looks safe? If yes, we can get rid of `readyTransactionId` variable.

Reply:
It should certainly be safe to get rid of the `readyTransactionId`. And we can probably also remove the `setTransactionId(..)` method. Or is it being used for something else?

Reply:
`setTransactionId(...)` sets the `transactionId` field in `TransactionContextImpl` 's builder class. This value is further used within the constructor of `TransactionContextImpl` to set member `transactionId`. And I see that its further used by other classes.

So it will be best to retain the setter.